.github/workflows/build-and-test-toolchain.yml

```diff
@@ -155,6 +160,10 @@ jobs:
         run: |
           cat ${{ steps.get-wsl-paths.outputs.artifact-path }}\gcc-tests-${{ env.TAG }}\summary.txt >> $env:GITHUB_STEP_SUMMARY
 
+      - name: Group failing tests by filepath of their source files
+        run: |
+          python .github/scripts/toolchain/group-gcc-results-by-source.py $ARTIFACT_PATH/gcc-tests-${{ env.TAG }} >> $ARTIFACT_PATH/gcc-tests-${{ env.TAG }}/grouped-failing-by-source.txt
```

Comment:
What about adding the, let's say, 10 topmost failures to the `$env:GITHUB_STEP_SUMMARY`?

Reply:
Not sure if 10 topmost failures would be useful. They don't need to be the 10 most frequent. E.g. in the following it would show `dup_s16.c: 476` etc. but not `sme2: 46764` and further

```
gcc: 218064
  target: 214927
    aarch64: 214927
      sve: 112559
        acle: 111480
          asm: 110992
            dup_s16.c: 476
            dup_u16.c: 476
            dup_s32.c: 460
            dup_s64.c: 460
            dup_u32.c: 460
            dup_u64.c: 460
            cvt_f16.c: 300
            ...
      sme2: 46764
        acle-asm: 46764
          ...
      sve2: 45601
        acle: 45596
          asm: 45596
            ...
      sme: 7410
        acle-asm: 7238
          ...
```

Reply:
what about top 10 by first level grouping?
something like this
```
gcc: 218064
  target: 214927
    aarch64: 214927
      sve: 112559
        acle: 111480
          asm: 110992
      sme2: 46764
        acle-asm: 46764
      sve2: 45601
        acle: 45596
          asm: 45596
      sme: 7410
```

Reply:
Sure, 10 most frequent issues was what I meant. I was not that explicit, sorry.